content/docs/connect/connection-pooling.md

```diff
@@ -63,7 +63,9 @@ Even with the largest compute size, the `max_connections` limit may not be suffi
 Some applications open numerous connections, with most eventually becoming inactive. This behavior can often be attributed to database driver limitations, running many instances of an application, or applications with serverless functions. With regular Postgres, new connections are rejected when reaching the `max_connections` limit. To overcome this limitation, Neon supports connection pooling using [PgBouncer](https://www.pgbouncer.org/), which allows Neon to support up to 10,000 concurrent connections.
 
 <Admonition type="important">
-Connection pooling supports up to 10,000 concurrent connections when multiple Postgres users are connecting to multiple Postgres databases. For a single user connecting to a single database, the maximum number of concurrent connections is 64. This limit is controlled by the PgBouncer `default_pool_size` parameter, which defines the default number of server connections per user/database pair. See [Neon PgBouncer configuration settings](#neon-pgbouncer-configuration-settings). If your application connects to your database using a single Postgres role, expect to encounter this limit when exceeding the 64 concurrent connection limit. For information about creating additional Postgres roles, see [Manage roles](/docs/manage/roles#manage-roles-with-sql). Roles can created via the Neon Console, CLI, API, and SQL.
+Connection pooling supports up to 10,000 concurrent connections. However, these connections are not a magic bullet: These connections are pooled and take shared ownership of a pool of connections to PostgreSQL. This pool is shared between all connections that connect with the same user credentials to the same database, and share a total of up to [64 connections](#neon-pgbouncer-configuration-settings). This means any user can only have 64 concurrently active transactions in a single database when connected through the pooler endpoint.  
+Additionally, because PgBouncer still needs to connect to PostgreSQL, the number of active concurrent connections is further limited by `max_connections`: if the `max_connections` parameter of Postgres is 112 (i.e. the endpoint was started with 0.25 CU), you can still hit PostgreSQL's own connection limit with only 2 roles.  
```

Comment:
If the total limit here is max_connections regardless of Pooling, per min 8 CU: https://neon.tech/docs/connect/connection-pooling#connection-limits-without-connection-pooling it is upto 3604 connections.

But I know what you mean, the client can still open 10K - the Pgbouncer can open upto max_connections with pg. can we make that more clear here that here we're talking about pgbouncer to pg (a system inside of Neon?)?